clang/include/clang/Basic/TargetInfo.h

```diff
@@ -1780,6 +1780,14 @@ class TargetInfo : public TransferrableTargetInfo,
     return 0;
   }
 
+  /// \returns Target specific address space for indirect (e.g. sret) arguments.
+  /// If such an address space exists, it must be convertible to and from the
+  /// alloca address space. If it does not, std::nullopt is returned and the
+  /// alloca address space will be used.
+  virtual std::optional<unsigned> getIndirectArgAddressSpace() const {
```

Comment:
I would expect this to be a `LangAS`, since that's what our address-space conversion lowerings are generally expressed in terms of.  This also has the advantage of avoiding a *lot* of heartache with implicit conversions around `ABIInfo::getIndirect`, since `LangAS` is a scoped enum.  And `LangAS::Default` is a much more reasonable default argument for things like `ABIArgInfo::getIndirect` than IR addrspace 0.

Reply:
This also shouldn't be optional. There always must be a definitive IR address space.

Also I'm not sure I follow why this is still necessary if you've modified getIndirect to carry the address space. RetAI should have this info now?

Reply:
Err, I forgot to delete this. RE: `unsigned` vs `LangAS`, I used the former for symmetry with other interfaces. I agree with @rjmccall that `LangAS` would be safer / would make more sense, however `getIndirectAliased` already uses numeric vs typed.